index.html

```diff
@@ -0,0 +1,89 @@
+<!doctype html>
+<!--
+Copyright 2018 The Go Authors. All rights reserved.
+Use of this source code is governed by a BSD-style
+license that can be found in the LICENSE file.
+-->
+<html>
+
+<head>
+	<meta charset="utf-8">
+	<title>Go wasm</title>
+</head>
+
+<body>
+	<!--
+	Add the following polyfill for Microsoft Edge 17/18 support:
+	<script src="https://cdn.jsdelivr.net/npm/text-encoding@0.7.0/lib/encoding.min.js"></script>
+	(see https://caniuse.com/#feat=textencoder)
+	-->
+	<script src="wasm_exec.js"></script>
+	<script>
+		if (!WebAssembly.instantiateStreaming) { // polyfill
+			WebAssembly.instantiateStreaming = async (resp, importObject) => {
+				const source = await (await resp).arrayBuffer();
+				return await WebAssembly.instantiate(source, importObject);
+			};
+		}
+
+		let exitCode = 0;
+		function goExit(code) {
+			exitCode = code;
+		}
+		function enosys() {
+			const err = new Error("not implemented");
+			err.code = "ENOSYS";
+			return err;
+		}
+		let coverageProfileContents;
+
+		(async() => {
+			const go = new Go();
+			const defaultWriteSync = window.fs.writeSync;
+			const coverFileDescriptor = 3; // 1 above standard descriptors (0-2)
+			const coverFilePath = {{.CoverageFile}};
+			window.fs.open = (path, flags, mode, callback) => {
```

Comment:
I'd prefer to move this whole `globalThis.fs` overriding business to a separate function for clarity:

```js
let coverageProfileContents;
function overrideFs(fs) {
	const defaultWriteSync = fs.writeSync;
	const coverFilePath = {{.CoverageFile}};
	const coverFileDescriptor = 3; // 1 above standard descriptors (0-2)
	fs.open = (path, flags, mode, callback) => {
		if (path !== coverFilePath) {
			callback(enosys());
			return;
		}
		coverageProfileContents = "";
		callback(null, coverFileDescriptor);
	};
	fs.close = (fd, callback) => {
		if (fd !== coverFileDescriptor) {
			callback(enosys());
			return;
		}
		callback(null);
	};
	const decoder = new TextDecoder("utf-8");
	fs.writeSync = (fd, buf) => {
		if (fd === coverFileDescriptor) {
			coverageProfileContents += decoder.decode(buf);
			return buf.length;
		}
		return defaultWriteSync(fd, buf);
	};
	return fs
}
```

And inside the async function, just call `globalThis.fs = overrideFs(globalThis.fs);`. Slightly more readable that way.

Reply:
Fixed in [722a2b6](https://github.com/agnivade/wasmbrowsertest/pull/41/commits/722a2b6cc6ad62e87cc275a0e906b7265830aaf7)